tests/test_db.rs

```diff
@@ -0,0 +1,183 @@
+  assert!(db.get(b"k1").unwrap().is_none());
+}
+
+#[test]
+fn errors_do_stuff() {
+    let path = DBPath::new("_rust_rocksdb_error");
+    let _db = DB::open_default(&path).unwrap();
+    let opts = Options::default();
+    // The DB will still be open when we try to destroy it and the lock should fail.
+    match DB::destroy(&opts, &path) {
+        Err(s) => {
+            let message = s.to_string();
+            assert!(message.find("IO error:").is_some());
+            assert!(message.find("_rust_rocksdb_error").is_some());
+            assert!(message.find("/LOCK:").is_some());
+        }
+        Ok(_) => panic!("should fail"),
+    }
+}
+
+#[test]
+fn writebatch_works() {
+    let path = DBPath::new("_rust_rocksdb_writebacktest");
+    {
+        let db = DB::open_default(&path).unwrap();
+        {
+            // test put
+            let mut batch = WriteBatch::default();
+            assert!(db.get(b"k1").unwrap().is_none());
+            assert_eq!(batch.len(), 0);
+            assert!(batch.is_empty());
+            let _ = batch.put(b"k1", b"v1111");
+            assert_eq!(batch.len(), 1);
+            assert!(!batch.is_empty());
+            assert!(db.get(b"k1").unwrap().is_none());
+            let p = db.write(batch);
+            assert!(p.is_ok());
+            let r: Result<Option<DBVector>, Error> = db.get(b"k1");
+            assert!(r.unwrap().unwrap().to_utf8().unwrap() == "v1111");
+        }
+        {
+            // test delete
+            let mut batch = WriteBatch::default();
+            let _ = batch.delete(b"k1");
+            assert_eq!(batch.len(), 1);
+            assert!(!batch.is_empty());
+            let p = db.write(batch);
+            assert!(p.is_ok());
+            assert!(db.get(b"k1").unwrap().is_none());
+        }
+        {
+            // test size_in_bytes
+            let mut batch = WriteBatch::default();
+            let before = batch.size_in_bytes();
+            let _ = batch.put(b"k1", b"v1234567890");
+            let after = batch.size_in_bytes();
+            assert!(before + 10 <= after);
+        }
+    }
+}
+
+#[test]
+fn iterator_test() {
+    let path = DBPath::new("_rust_rocksdb_iteratortest");
+    {
+        let db = DB::open_default(&path).unwrap();
+        let p = db.put(b"k1", b"v1111");
+        assert!(p.is_ok());
+        let p = db.put(b"k2", b"v2222");
+        assert!(p.is_ok());
+        let p = db.put(b"k3", b"v3333");
+        assert!(p.is_ok());
+        let iter = db.iterator(IteratorMode::Start);
+        for (k, v) in iter {
+            println!(
+                "Hello {}: {}",
+                str::from_utf8(&*k).unwrap(),
+                str::from_utf8(&*v).unwrap()
+            );
+        }
+    }
+}
+
+#[test]
+fn snapshot_test() {
+    let path = DBPath::new("_rust_rocksdb_snapshottest");
+    {
+        let db = DB::open_default(&path).unwrap();
+        let p = db.put(b"k1", b"v1111");
+        assert!(p.is_ok());
+
```

Comment:
Do we really need `let p ...` here ? 
```suggestion
        assert!(db.put(b"k1", b"v1111").is_ok());
```